### PR TITLE
chore(docs): queue single consumer

### PR DESCRIPTION
**Pull request by ekeren**
## Checklist

- [ ] Title matches [Winglang's style guide](https://docs.winglang.io/contributors/pull_requests#how-are-pull-request-titles-formatted)
- [ ] Description explains motivation and solution
- [ ] Tests added (always)
- [ ] Docs updated (only required for features)
- [ ] Added `pr/e2e-full` label if this feature requires end-to-end testing

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Monada Contribution License](https://docs.winglang.io/terms-and-policies/contribution-license.html)*.

### CODE REVIEW

File: docs/05-reference/wingsdk-spec.md [hunk 1]
```diff
@@ -407,14 +407,9 @@ Future extensions:
 > - What is the API for message deduplication?
 > - Should the API use "push"/"pop" terminology or "send"/"receive" terminology?
 
-The queue resource represents a message buffer that can be used to decouple workloads between a set of producers and a set of consumers.
-Using a queue, you can send, store, and receive messages between software components at any volume, without losing messages or requiring other services to be available.
+A Queue represents a data structure that holds a list of items. The distinguishing characteristic of a Queue in our API is its design to interact with a single consumer service.
```
Comment (ekeren) [inline docs/05-reference/wingsdk-spec.md:410]:
```suggestion
A Queue represents a data structure that holds a list of items. Using a queue, you can send, store, and receive messages between software components at any volume, without losing messages or requiring other services to be available.

Any number of producers can push messages to the queue, and a single consumer can pop messages from the queue.
```

**Comment by ShaiBer** [comment]
Shouldn't we also change the api from addConsumer to setConsumer and make it clear that subsequent calls will replace the consumer?

**Comment by ekeren** [comment]
> Shouldn't we also change the api from addConsumer to setConsumer and make it clear that subsequent calls will replace the consumer?

 @eladb ?

**Comment by eladb** [comment]
> > Shouldn't we also change the api from addConsumer to setConsumer and make it clear that subsequent calls will replace the consumer?
> 
> @eladb ?

The philosophy behind using "addXxx" is a bit of a purist approach we had in the design of the CDK APIs where you basically "add" the consumer to the definition of the queue.

I am okay revisiting this idea and changing to `setCustomer`.

@Chriscbr any objections?

File: docs/05-reference/wingsdk-spec.md [hunk 2]
```diff
@@ -407,14 +407,11 @@ Future extensions:
...
 > - Should the API use "push"/"pop" terminology or "send"/"receive" terminology?
 
-The queue resource represents a message buffer that can be used to decouple workloads between a set of producers and a set of consumers.
-Using a queue, you can send, store, and receive messages between software components at any volume, without losing messages or requiring other services to be available.
+A Queue represents a data structure that holds a list of items. Using a queue, you can send, store, and receive messages between software components at any volume, without losing messages or requiring other services to be available.
 
-Any number of producers can push messages to the queue, and any number of consumers can pop messages from the queue.
-When a consumer function receives a batch of messages, it is responsible for processing each message in the batch and acknowledging that the message has been processed by deleting it from the queue.
-If a consumer function does not acknowledge a message, the message will be re-delivered to another consumer function.
+Any number of producers can push messages to the queue, and a single consumer can pop messages from the queue.
```
[the remark this answers is not in the document]
Comment (ekeren) [reply docs/05-reference/wingsdk-spec.md:412]: I agree. (forgot that we also have the pop API, and not only the addConsumer)

**Pull request merged**